### PR TITLE
MET-4472 New Tier Calculation for 3D

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

**Review by stzanakis — CHANGES_REQUESTED** [review]

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java [hunk 1]
```diff
@@ -30,8 +33,24 @@ MediaTier classifyEntityWithoutWebResources(RdfWrapper entity, boolean hasLandin
   @Override
   MediaTier classifyWebResource(WebResourceWrapper webResource, boolean hasLandingPage, boolean hasEmbeddableMedia) {
 
-    // T2-T4 if there is a mime type (any whatsoever), T0 otherwise.
-    return StringUtils.isNotBlank(webResource.getMimeType()) ? MediaTier.T4 : MediaTier.T0;
+    if(webResource == null){
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java:36]: Changed it

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java [hunk 2]
```diff
@@ -30,8 +33,24 @@ MediaTier classifyEntityWithoutWebResources(RdfWrapper entity, boolean hasLandin
...
+      return MediaTier.T0;
+    }
+
+    MediaTier result = MediaTier.T0;
+
+    if(hasMediaResourceThatIsNotImageOrPdf(webResource) && hasLicenseType(webResource,LicenseType.OPEN)){
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java:42]: I see what you mean, I will remove the license evaluation part. It was mentioned in the document which tier to gives based on the type of license as well, AbstractMediaClassifier already handles it then?

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java [hunk 3]
```diff
@@ -43,4 +62,23 @@ ResolutionTierMetadata extractResolutionTierMetadata(WebResourceWrapper webResou
   MediaType getMediaType() {
     return MediaType.THREE_D;
   }
+
+  private boolean hasMediaResourceThatIsNotImageOrPdf(WebResourceWrapper webResource){
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java:66]: Just separated them

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java [hunk 4]
```diff
@@ -43,4 +62,23 @@ ResolutionTierMetadata extractResolutionTierMetadata(WebResourceWrapper webResou
...
+    Set<WebResourceLinkType> extractedLinkTypes = webResource.getLinkTypes();
+    String mimeType = webResource.getMimeType();
+    return extractedLinkTypes != null && mimeType != null && (extractedLinkTypes.contains(WebResourceLinkType.IS_SHOWN_BY) ||
+            extractedLinkTypes.contains(WebResourceLinkType.HAS_VIEW)) && !mimeType.contains("image") &&
+            !mimeType.contains("application/pdf");
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java:71]: Just saw the check in TextClassifier and re-used it here

File: metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java [hunk 5]
```diff
@@ -30,8 +32,19 @@ MediaTier classifyEntityWithoutWebResources(RdfWrapper entity, boolean hasLandin
   @Override
   MediaTier classifyWebResource(WebResourceWrapper webResource, boolean hasLandingPage, boolean hasEmbeddableMedia) {
```
Comment (stzanakis) [inline metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/ThreeDClassifier.java:33]: Looks good now!

**Pull request merged**